sys/random/mersenne.c

```diff
@@ -129,7 +133,12 @@ uint32_t genrand_uint32(void)
     y ^= (y << 7) & 0x9d2c5680UL;
     y ^= (y << 15) & 0xefc60000UL;
     y ^= y >> 18;
+
+#ifdef MODULE_ENTROPY
+    return y + entropy;
```

Comment:
Speaking as a mathematician: Entropy is not randomness, but something like 'unpredictability'. It is mostly needed for crypto.
It is the continuous experience that adding something 'that looks random' is rather harmful than helpful. A colleague, who is a crypto-mathematician, just recently told me that (with NXP) they had performed a larger study on common sources of 'entropy' (typing, perturbed wireless signals, other hardware fluctuations) and results were disastrous: no reliable unpredictability nowhere. 

Practically speaking, a pseudo-random number generator basically reproduces certain statistical properties, starting it's 'randomness'  from initialization (seeds). With equal seeds and common generators, number sequences deterministically repeat, which often is a desired property for debugging.  I would stick to this approach and clearly state that this is not crypto-safe. 
Adding 'something' to the number output, mainly produces the threat of spoiling the statistical properties ... it is very unlikely to maximize entropy by this.